Error while starting using interactsh server

When start using Interactsh Server: oast.me
an error pops up:

```bash
panic: assignment to entry in nil map

goroutine 243436 [running]:
github.com/projectdiscovery/nuclei/v3/pkg/templates.(*ClusterExecuter).Execute.func1(0xc0167b5a90)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/templates/cluster.go:257 +0x1a6
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults.func1.1(0xc0167b5a90)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/protocols/http/request.go:436 +0x435
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).executeRequest(0xc01164b4a0, 0xc01234a990, 0xc00ec6fae0, 0xc007320c90, 0x0, 0xc008699b18, 0x0?)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/protocols/http/request.go:706 +0x3562
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults.func1({0xc00e34d500, 0x1c}, 0x7f4f28685bc8?, 0x41f656?)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/protocols/http/request.go:409 +0x56b
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults(0xc01164b4a0, 0xc01234a990, 0xc007320cc0, 0xc007320c90, 0xc01234a9a8)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/protocols/http/request.go:473 +0x36a
github.com/projectdiscovery/nuclei/v3/pkg/templates.(*ClusterExecuter).Execute(0xc0049b5340, 0xc012024000)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/templates/cluster.go:254 +0x14a
github.com/projectdiscovery/nuclei/v3/pkg/core.(*Engine).executeTemplateWithTargets.func2.1(0x7c5405?, 0x0?, 0xc01111d100)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/core/executors.go:132 +0x1d8
created by github.com/projectdiscovery/nuclei/v3/pkg/core.(*Engine).executeTemplateWithTargets.func2 in goroutine 243434
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.1/pkg/core/executors.go:108 +0x459
```

--- a/pkg/protocols/http/request.go
+++ b/pkg/protocols/http/request.go
@@ -503,6 +503,7 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 				event.InternalEvent = outputEvent
 			}
 			callback(event)
+			return
 		}
 		if event.InternalEvent == nil {
 			event.InternalEvent = outputEvent

--- a/pkg/protocols/http/request.go
+++ b/pkg/protocols/http/request.go
@@ -488,6 +488,22 @@ var errStopExecution = errors.New("stop execution due to unresolved variables")
 
 // executeRequest executes the actual generated request and returns error if occurred
 func (request *Request) executeRequest(input *contextargs.Context, generatedRequest *generatedRequest, previousEvent output.InternalEvent, hasInteractMatchers bool, callback protocols.OutputEventCallback, requestCount int) error {
+	outputEvent := output.InternalEvent{}
+	var event *output.InternalWrappedEvent
+	// event should never be nil as per existing logic
+	defer func() {
+		if event == nil {
+			event := &output.InternalWrappedEvent{}
+			if request.CompiledOperators != nil && request.CompiledOperators.HasDSL() {
+				event.InternalEvent = outputEvent
+			}
+			callback(event)
+		}
+		if event.InternalEvent == nil {
+			event.InternalEvent = outputEvent
+		}
+	}()
+
 	request.setCustomHeaders(generatedRequest)
 
 	// Try to evaluate any payloads before replacement
@@ -688,7 +704,7 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 		// In case of interactsh markers and request times out, still send
 		// a callback event so in case we receive an interaction, correlation is possible.
 		// Also, to log failed use-cases.
-		outputEvent := request.responseToDSLMap(&http.Response{}, input.MetaInput.Input, formedURL, convUtil.String(dumpedRequest), "", "", "", 0, generatedRequest.meta)
+		outputEvent = request.responseToDSLMap(&http.Response{}, input.MetaInput.Input, formedURL, convUtil.String(dumpedRequest), "", "", "", 0, generatedRequest.meta)
 		if i := strings.LastIndex(hostname, ":"); i != -1 {
 			hostname = hostname[:i]
 		}
@@ -698,12 +714,6 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 		} else {
 			outputEvent["ip"] = httpclientpool.Dialer.GetDialedIP(hostname)
 		}
-
-		event := &output.InternalWrappedEvent{}
-		if request.CompiledOperators != nil && request.CompiledOperators.HasDSL() {
-			event.InternalEvent = outputEvent
-		}
-		callback(event)
 		return err
 	}
 
@@ -775,7 +785,7 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 		}
 		finalEvent := make(output.InternalEvent)
 
-		outputEvent := request.responseToDSLMap(respChain.Response(), input.MetaInput.Input, matchedURL, convUtil.String(dumpedRequest), respChain.FullResponse().String(), respChain.Body().String(), respChain.Headers().String(), duration, generatedRequest.meta)
+		outputEvent = request.responseToDSLMap(respChain.Response(), input.MetaInput.Input, matchedURL, convUtil.String(dumpedRequest), respChain.FullResponse().String(), respChain.Body().String(), respChain.Headers().String(), duration, generatedRequest.meta)
 		// add response fields to template context and merge templatectx variables to output event
 		request.options.AddTemplateVars(input.MetaInput, request.Type(), request.ID, outputEvent)
 		if request.options.HasTemplateCtx(input.MetaInput) {
@@ -813,7 +823,7 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 		// prune signature internal values if any
 		request.pruneSignatureInternalValues(generatedRequest.meta)
 
-		event := eventcreator.CreateEventWithAdditionalOptions(request, generators.MergeMaps(generatedRequest.dynamicValues, finalEvent), request.options.Options.Debug || request.options.Options.DebugResponse, func(internalWrappedEvent *output.InternalWrappedEvent) {
+		event = eventcreator.CreateEventWithAdditionalOptions(request, generators.MergeMaps(generatedRequest.dynamicValues, finalEvent), request.options.Options.Debug || request.options.Options.DebugResponse, func(internalWrappedEvent *output.InternalWrappedEvent) {
 			internalWrappedEvent.OperatorsResult.PayloadValues = generatedRequest.meta
 		})
 		if hasInteractMatchers {

--- a/pkg/protocols/http/request.go
+++ b/pkg/protocols/http/request.go
@@ -493,7 +493,12 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 	// event should never be nil as per existing logic
 	defer func() {
 		if event == nil {
-			event := &output.InternalWrappedEvent{}
+			event := &output.InternalWrappedEvent{
+				InternalEvent: map[string]interface{}{
+					"template-id": request.options.TemplateID,
+					"host":        input.MetaInput.Input,
+				},
+			}
 			if request.CompiledOperators != nil && request.CompiledOperators.HasDSL() {
 				event.InternalEvent = outputEvent
 			}
@@ -501,6 +506,13 @@ func (request *Request) executeRequest(input *contextargs.Context, generatedRequ
 		}
 		if event.InternalEvent == nil {
 			event.InternalEvent = outputEvent
+		}
+		// make sure templateId is never nil
+		if event.InternalEvent["template-id"] == nil {
+			event.InternalEvent["template-id"] = request.options.TemplateID
+		}
+		if event.InternalEvent["host"] == nil {
+			event.InternalEvent["host"] = input.MetaInput.Input
 		}
 	}()
 

--- a/pkg/templates/cluster.go
+++ b/pkg/templates/cluster.go
@@ -252,6 +252,13 @@ func (e *ClusterExecuter) Execute(ctx *scan.ScanContext) (bool, error) {
 	previous := make(map[string]interface{})
 	dynamicValues := make(map[string]interface{})
 	err := e.requests.ExecuteWithResults(inputItem, dynamicValues, previous, func(event *output.InternalWrappedEvent) {
+		if event == nil {
+			// unlikely but just in case
+			return
+		}
+		if event.InternalEvent == nil {
+			event.InternalEvent = make(map[string]interface{})
+		}
 		for _, operator := range e.operators {
 			result, matched := operator.operator.Execute(event.InternalEvent, e.requests.Match, e.requests.Extract, e.options.Options.Debug || e.options.Options.DebugResponse)
 			event.InternalEvent["template-id"] = operator.templateID

